archaius2-core/src/main/java/com/netflix/archaius/ConfigProxyFactory.java

```diff
@@ -460,6 +485,19 @@ private static void maybeWrapThenRethrow(Throwable t) {
         throw new RuntimeException(t);
     }
 
+    private static <T> void warnWhenTooMany(Map<T, Integer> counters, T countKey, int limit, Supplier<String> objectDescription) {
+        int currentCount = counters.merge(countKey, 1, Integer::sum);
+
+        // Emit warning if we're over the limit BUT only when the current count is a multiple of the limit :-)
+        // This is to avoid being *too* noisy
+        if (LOG.isWarnEnabled() &&
+            currentCount >= limit &&
+            (currentCount % limit == 0 )) {
+
+                LOG.warn("Too many {} objects are being created. Please review the calling code to prevent memory leaks. Stack trace for debugging follows:",
```

Comment:
Nit: Perhaps log the `EXCESSIVE_PROXIES_LIMIT` knob so users know how to adjust this if they have a legit reason to create so many of these proxies.  But also log that you should only be creating a singleton.

Reply:
Done :-)
New log message looks like
```
2023-10-30 13:30:48,612 WARN  ConfigProxyFactory:497  Too many ProxyFactory(Config:330084561) objects are being created (5 so far).
Please review the calling code to prevent memory leaks.
Normal usage for ConfigProxyFactory is to create singletons via your DI mechanism.
For special use cases that *require* creating multiple instances you can tune reporting
by setting the `archaius.excessiveProxiesLogging.limit` config key to a higher threshold.
Stack trace for debugging follows: [Test worker]
java.lang.Throwable
	at com.netflix.archaius.ConfigProxyFactory.warnWhenTooMany(ConfigProxyFactory.java:503)
	at com.netflix.archaius.ConfigProxyFactory.<init>(ConfigProxyFactory.java:146)
	at com.netflix.archaius.ProxyFactoryTest.testLogExcessiveUse(ProxyFactoryTest.java:563)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        ...
```